thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/rootcause/impl/DimensionPipeline.java

```diff
@@ -8,103 +10,142 @@
 import com.linkedin.thirdeye.datalayer.bao.MetricConfigManager;
 import com.linkedin.thirdeye.datalayer.dto.DatasetConfigDTO;
 import com.linkedin.thirdeye.datalayer.dto.MetricConfigDTO;
-import com.linkedin.thirdeye.rootcause.ExecutionContext;
 import com.linkedin.thirdeye.rootcause.Pipeline;
+import com.linkedin.thirdeye.rootcause.PipelineContext;
 import com.linkedin.thirdeye.rootcause.PipelineResult;
-import java.util.ArrayList;
-import java.util.List;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
 import java.util.Set;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 
 /**
- * Sample implementation of a pipeline for identifying relevant dimensions by performing
+ * Pipeline for identifying relevant dimensions by performing
  * contribution analysis. The pipeline first fetches the Current and Baseline entities and
  * MetricEntities in the search context. It then maps the metrics to ThirdEye's internal database
  * and performs contribution analysis using a {@code DimensionScorer).
  *
  * @see DimensionScorer
  */
-public class DimensionPipeline implements Pipeline {
+public class DimensionPipeline extends Pipeline {
```

Comment:
Why not call it ContributionAnalysisPipeline or DimensionContributionAnalysisPipeline or DimensionAnalysisPipeline